Add more warehouses to database permissions

agnes-kiss opened the pull request:
## What are you changing in this pull request and why?
The proposed changes are regarding this issue raised last week: https://github.com/dbt-labs/docs.getdbt.com/issues/3901

It would be nice to have more warehouses within References / warehouse permissions with sample grants, this PR aims to follow the already existing [Snowflake](https://docs.getdbt.com/reference/snowflake-permissions) example and add Databricks, Redshift, and Postgres as well (skipping BigQuery due to their unique IAM access control management)

#Resolves #3824

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [x] Provide a unique filename for the new page

- [x] Needs to have a technical review, at least to verify the proposed changes are correct and possibly expand with other examples.
- [x] Tested it locally (npm run)

agnes-kiss committed:
Add more warehouses to database permissions

agnes-kiss committed:
Fix broken links

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 committed:
Merge branch 'current' into database_permissions

rlh1994 commented:
Hey @mirnawong1 is there anything we need to change to be able to get this merged please?

mirnawong1 commented on website/docs/reference/database-permissions/databricks-permissions.md:
````diff
@@ -0,0 +1,17 @@
+grant create view on schema schema_name to principal;
+```
+
+Check out the [Official Documentation](https://docs.databricks.com/en/data-governance/unity-catalog/manage-privileges/privileges.html#privilege-types-by-securable-object-in-unity-catalog) for more information.
````
```suggestion
Check out the [official documentation](https://docs.databricks.com/en/data-governance/unity-catalog/manage-privileges/privileges.html#privilege-types-by-securable-object-in-unity-catalog) for more information.
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/databricks-permissions.md

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 commented:
this is so great @agnessnowplow and @rlh1994 ! thanks so much for opening this up and providing such diligent and detailed contributions 😍 ! apologies for the delay here and I'll review this right now - am going to loop in @dataders as well for his technical help - thank you for flagging @rlh1994 !

mirnawong1 commented on website/docs/reference/database-permissions/redshift-permissions.md:
````diff
@@ -0,0 +1,18 @@
+grant select on all views in database database_name to user_name;
+```
+
+Check out the [Official Documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_GRANT.html) for more information.
````
```suggestion
Check out the [official documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_GRANT.html) for more information.
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/redshift-permissions.md

mirnawong1 commented on website/docs/reference/database-permissions/postgres-permissions.md:
````diff
@@ -0,0 +1,18 @@
+grant select on all views in database database_name to user_name;
+```
+
+Check out the [Official Documentation](https://www.postgresql.org/docs/current/sql-grant.html) for more information.
````
```suggestion
Check out the [official documentation](https://www.postgresql.org/docs/current/sql-grant.html) for more information.
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/postgres-permissions.md

mirnawong1 commented on website/docs/reference/references-overview.md:
```diff
@@ -51,7 +51,25 @@ Learn how to add more configurations to your dbt project or adapter, use propert
 <Card
     title="Snowflake Permissions Artifacts"
```
```suggestion
    title="Snowflake permissions artifacts"
```

mirnawong1 committed:
Update website/docs/reference/references-overview.md

mirnawong1 commented on website/docs/reference/references-overview.md:
```diff
@@ -49,9 +49,27 @@ Learn how to add more configurations to your dbt project or adapter, use propert
+    icon="computer"/>
+
+<Card
+    title="Databricks Permissions Artifacts"
```
```suggestion
    title="Databricks permissions artifacts"
```

mirnawong1 committed:
Update website/docs/reference/references-overview.md

mirnawong1 commented on website/docs/reference/references-overview.md:
```diff
@@ -49,9 +49,27 @@ Learn how to add more configurations to your dbt project or adapter, use propert
+    icon="computer"/>
+
+<Card
+    title="Postgres Permissions Artifacts"
```
```suggestion
    title="Postgres permissions artifacts"
```

mirnawong1 committed:
Update website/docs/reference/references-overview.md

mirnawong1 commented on website/docs/reference/references-overview.md:
```diff
@@ -49,9 +49,27 @@ Learn how to add more configurations to your dbt project or adapter, use propert
+    icon="computer"/>
+
+<Card
+    title="Redshift Permissions Artifacts"
```
```suggestion
    title="Redshift permissions artifacts"
```

mirnawong1 committed:
Update website/docs/reference/references-overview.md

mirnawong1 commented:
hey @dataders , i love how detailed this pr is and think it'll help users. i wonder if you can help me add some contextual info on how these permissions may be helpful for users? also do you think i should link to them from the dbt cloud data platform pages (e.g. databricks, redshift, etc.) like the [snowflake page(https://docs.getdbt.com/docs/cloud/connect-data-platform/connect-snowflake) already does?

runleonarun committed:
Merge branch 'current' into database_permissions

rlh1994 commented:
Hey @mirnawong1 and @dataders just wanted to check if there was any update on this please?

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 commented:
hey @rlh1994 apologies for the delay here. I'm going to look into this this week so we can finalize this pr for you! thank you for the flag!

dataders reviewed (commented):
I'm in support of these contributions. Below I have questions. @mirnawong1 I leave it up to you though how you want this merged. This PR is certainly an improvement to what we have currently, but it feels half done still.

### questions

1. the purpose is to call out permissions for analytics engineers / developers not end users of these tables? if so this should be made clear
2. i'm not sure this reference section should exist, perhaps instead it should be a new section in platform-specific the `*-configs.md` guides?
3. should the other `connect-*.md` guides also call out the the permissions sections as snowflake's does?
4. can we see more consistency between the pages. for example
  1. the `NOTE` is present in some and not others
  2. Snowflake's grant's database permissions and others do not
  3. databrick's page links to official documentation, others do not

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 commented:
thanks @dataders , i'm going to review this tomorrow - appreciate it!

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 committed:
Create about-database-permissions.md

add landing page

mirnawong1 committed:
Update about-database-permissions.md

mirnawong1 committed:
Update snowflake-permissions.md

mirnawong1 committed:
Update databricks-permissions.md

mirnawong1 committed:
Update databricks-permissions.md

mirnawong1 committed:
Update snowflake-permissions.md

rlh1994 commented on website/docs/reference/database-permissions/about-database-permissions.md:
```diff
@@ -0,0 +1,26 @@
+- They help maintain data integrity, prevent unauthorized changes, and limit exposure to sensitive data.
+- Permissions also support compliance with data privacy regulations and auditing.
+
+## How to use them
```
Is it worth adding here that permissions can be granted at various levels e.g. table, schema, warehouses, database etc. Also that each cloud vendor has a slightly different approach and naming to privelages so to check each vendors documentation for more specifics?

mirnawong1 replied:
great flag @rlh1994 , i've added some more context based on your sugg above!

mirnawong1 committed:
Update postgres-permissions.md

rlh1994 commented on website/docs/reference/database-permissions/postgres-permissions.md:
```diff
@@ -2,8 +2,15 @@
+
 ## Example Postgres permissions
 
+The following example provides you with the SQL statements you can use to manage permissions. 
```
From memory (@agnessnowplow correct me if I'm wrong) these ones were specifically chosen to allow the user to then run dbt without any permissions issues ( i.e. create schemas, read existing data, access the information schema). Could it be better to explain that here rather than list these as examples?

agnes-kiss replied:
@mirnawong1  Thank you for all the changes! Those permissions are definitely aimed at helping users run dbt without permission issues. I cannot not say it is a 100% full list but some of the most important ones for sure.

mirnawong1 committed:
Update redshift-permissions.md

rlh1994 commented on website/docs/reference/database-permissions/postgres-permissions.md:
```diff
@@ -2,8 +2,15 @@
 ---
 
+
+In Postgres, permissions are used to control who can perform certain actions on different database objects. Use SQL statements to manage permissions in a Databricks database.
```
```suggestion
In Postgres, permissions are used to control who can perform certain actions on different database objects. Use SQL statements to manage permissions in a Postgres database.
```

mirnawong1 committed:
Update redshift-permissions.md

mirnawong1 committed:
Update postgres-permissions.md

mirnawong1 committed:
Update databricks-permissions.md

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 commented on website/docs/reference/database-permissions/about-database-permissions.md:
```diff
@@ -0,0 +1,33 @@
+
+## How to use them
+
+- Users and administrators can grant and manage permissions using SQL statements or through the database system's interface.
```
```suggestion
- Users and administrators can grant and manage permissions at various levels (such as table, schema, and so on) using SQL statements or through the database system's interface.
```

mirnawong1 commented on website/docs/reference/database-permissions/about-database-permissions.md:
```diff
@@ -0,0 +1,33 @@
+
+## Examples
+
+For more info on examples and how to setup database permissions, refer to the following database permission pages:
```
```suggestion
Refer to the following database permission pages for more info on examples and how to setup database permissions:
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/about-database-permissions.md

mirnawong1 commented on website/docs/reference/database-permissions/about-database-permissions.md:
```diff
@@ -0,0 +1,33 @@
+- Assign permissions to individual users or roles (groups of users) based on their responsibilities.
+  - Typical permissions include "SELECT" (read), "INSERT" (add data), "UPDATE" (modify data), "DELETE" (remove data), and administrative rights like "CREATE" and "DROP."
+- Users should be assigned permissions that ensure they have the necessary access to perform their tasks without overextending privileges.
+
```
```suggestion

Something to note is that each data platform provider might have different approaches and names for privileges. Refer to their documentation for more details.
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/about-database-permissions.md

mirnawong1 commented on website/docs/reference/database-permissions/postgres-permissions.md:
```diff
@@ -0,0 +1,25 @@
+
+## Example Postgres permissions
+
+The following example provides you with the SQL statements you can use to manage permissions. 
```
how's this @rlh1994 ?

```suggestion
The following example provides you with the SQL statements you can use to manage permissions. These examples allow you to run dbt smoothly without encountering permission issues, such as creating schemas, reading existing data, and accessing the information schema. 
```

rlh1994 replied:
Looks good to me!

mirnawong1 commented on website/docs/reference/database-permissions/about-database-permissions.md:
```diff
@@ -0,0 +1,34 @@
+
+Database permissions are access rights and privileges granted to users or roles within a database or data platform. They help you specify what actions users or roles can perform on various database objects, like tables, views, schemas, or even the entire database.
+
+ARE THEY DIFF FROM MODEL GOVERNANCE BC CONTROLS THE DATABASE PERMISSIONS?
```
```suggestion
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/about-database-permissions.md

mirnawong1 committed:
Update sidebars.js

mirnawong1 committed:
Update about-database-permissions.md

mirnawong1 committed:
Update sidebars.js

mirnawong1 commented on website/sidebars.js:
```diff
@@ -936,7 +936,14 @@ const sidebarSettings = {
       type: "category",
       label: "Database Permissions",
-      items: ["reference/snowflake-permissions"],
+      link: { type: "doc", id: "reference/database-permissions/about-database-permissions" },
```
```suggestion
      link: { type: "doc", id: "reference/database-permissions/about-database-permissions" },
```

mirnawong1 commented on website/sidebars.js:
```diff
@@ -936,7 +936,14 @@ const sidebarSettings = {
-      items: ["reference/snowflake-permissions"],
+      link: { type: "doc", id: "reference/database-permissions/about-database-permissions" },
+      items: [
+	"reference/database-permissions/about-database-permissions"
```
```suggestion
	"reference/database-permissions/about-database-permissions"
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/postgres-permissions.md

mirnawong1 committed:
Update snowflake-permissions.md

adding discourse article to resolves issue https://github.com/dbt-labs/docs.getdbt.com/issues/3824

mirnawong1 committed:
Update sidebars.js

mirnawong1 commented on website/docs/faqs/Warehouse/database-privileges.md:
```diff
@@ -12,8 +12,8 @@ schema¹
-On Postgres, Redshift, and Snowflake, use a series of `grants` to ensure that
-your user has the correct privileges.
+On Postgres, Redshift, Databricks, and Snowflake, use a series of `grants` to ensure that
+your user has the correct privileges. Check out [example permissions](/reference/database-permissions/snowflake-permissions) for these warehouses.
```
```suggestion
your user has the correct privileges. Check out [example permissions](/reference/database-permissions/about-database-permissions) for these warehouses.
```

mirnawong1 committed:
Update website/docs/faqs/Warehouse/database-privileges.md

mirnawong1 committed:
Update connect-databricks.md

mirnawong1 committed:
Update connect-redshift-postgresql-alloydb.md

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 committed:
Update sidebars.js

mirnawong1 commented on website/sidebars.js:
```diff
@@ -936,7 +936,13 @@ const sidebarSettings = {
-      items: ["reference/snowflake-permissions"],
+      items: [
+	"reference/database-permissions/about-database-permissions",
+        "reference/database-permissions/databricks-permissions",
```
```suggestion
        "reference/database-permissions/databricks-permissions",
```

mirnawong1 committed:
Update sidebars.js

mirnawong1 committed:
Merge branch 'current' into database_permissions

mirnawong1 commented on website/docs/reference/database-permissions/snowflake-permissions.md:
```diff
@@ -0,0 +1,153 @@
+
+## Set up Snowflake account
+
+This section will explain the exact privileges it takes to set up your Snowflake account:
```
```suggestion 
This section explains how to set up permissions and roles within Snowflake. In Snowflake, you would perform these actions using SQL commands to set up your data warehouse and access control within Snowflake's ecosystem. 
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/snowflake-permissions.md

mirnawong1 commented on website/docs/reference/database-permissions/snowflake-permissions.md:
```diff
@@ -0,0 +1,153 @@
+
+## Set up Snowflake account
+
+This section explains how to set up permissions and roles within Snowflake. In Snowflake, you would perform these actions using SQL commands to set up your data warehouse and access control within Snowflake's ecosystem. 
```
```suggestion
This section explains how to set up permissions and roles within Snowflake. In Snowflake, you would perform these actions using SQL commands and set up your data warehouse and access control within Snowflake's ecosystem. 
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/snowflake-permissions.md

mirnawong1 commented on website/docs/reference/database-permissions/snowflake-permissions.md:
````diff
@@ -0,0 +1,153 @@
+```
+
+4. Create users, assigning them to their roles
+Every person and application gets a separate user and is assigned to the correct role.
````
```suggestion

Every person and application gets a separate user and is assigned to the correct role.
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/snowflake-permissions.md

mirnawong1 commented on website/docs/reference/database-permissions/about-database-permissions.md:
```diff
@@ -0,0 +1,35 @@
+Database permissions are access rights and privileges granted to users or roles within a database or data platform. They help you specify what actions users or roles can perform on various database objects, like tables, views, schemas, or even the entire database.
+
+
+## Why are they useful
```
```suggestion
### Why are they useful
```

mirnawong1 committed:
Update website/docs/reference/database-permissions/about-database-permissions.md

mirnawong1 committed:
Update about-database-permissions.md

mirnawong1 committed:
Update website/docs/reference/database-permissions/about-database-permissions.md

mirnawong1 reviewed (approved):
hey @agnessnowplow and @rlh1994 ! i think we're at a good place now and the [deploy preview](https://docs-getdbt-33t4rz1tw-dbt-labs.vercel.app/reference/database-permissions/about-database-permissions) looks pretty great! i really appreciate you opening this wonderful pr and flagging this for us - this page needed some attention and thanks to your contribution, it now has much more info. 

I'll get this merged for you but let me know if you have any questions or concerns
